shared/common-adapters/input.shared.tsx

```diff
@@ -2,6 +2,12 @@
 import {TextInfo} from './input'
 
 const checkTextInfo = ({text, selection}: TextInfo) => {
+  if (selection.end === null) {
+    throw new Error('selection.end null')
```

Comment:
why even make these nullable?

Reply:
its cause they are nullable. We could enforce they're not and catch it upstream but at the time it wasn't clear to me how many upstream vs downstream usages of it there were so i decided to let the type flow down so i could see. i think we likely do want to enforce it higher up. There is some code that assumes they're valid and i didn't want to test all those cases so this was simpler for now